cluster-autoscaler/core/scale_down.go

```diff
@@ -249,9 +263,17 @@ func (sd *ScaleDown) UpdateUnneededNodes(
 		}
 	}
 
-	// Update the timestamp map.
+	// Update the timestamp map. Look at empty and non empty nodes.
 	result := make(map[string]time.Time)
-	unneededNodesList := make([]*apiv1.Node, 0, len(nodesToRemove))
+	unneededNodesList := make([]*apiv1.Node, 0, len(nodesToRemove)+len(emptyNodes))
```

Comment:
could we merge emptyNods to nodesToRemove?

Reply:
Done.